pandas/_libs/tslibs/offsets.pyx

```diff
@@ -233,7 +233,7 @@ def _to_dt64D(dt):
     # numpy.datetime64('2013-05-01T02:00:00.000000+0200')
     # Thus astype is needed to cast datetime to datetime64[D]
     if getattr(dt, 'tzinfo', None) is not None:
-        i8 = pydt_to_i8(dt)
+        i8 = np.int64(dt.timestamp() * 10**9)
```

Comment:
Hmm I'm still a little hung up on this. Just to illustrate:

```python
dt1 = datetime.datetime(2020, 1, 17, 16, 1, 9, 136686)
>>> dt2 = pd.Timestamp(np.int64(dt1.timestamp() * 10**9))
>>> dt2.nanosecond
80
```

I suppose not totally equivalent  given the later cast to `datetime[D]` but this does lose some precision right? If so could lead to some hard to find bugs, but could also misunderstand

Reply:
I think you're right to be hung up on this, but for a different reason.  I had forgotten that the stdlib datetime.timestamp behaves different from pd.Timestamp.timestamp for tz-naive datetimes.  I'll give this another look.

Reply:
i take it back, i was right the first time.  this usage is only in the tzaware case, so we're all good

Reply:
shouldn't do what convet_datetime_to_tsobject does? this doesn't appear to have the same timezone behavior

Reply:
> this doesn't appear to have the same timezone behavior

On the previous line we are checking that we are only doing this in the tzaware case, in which datetime.timestamp and pd.Timestamp.timestamp behave the same.

Reply:
as i said i think you should re-use convert_datetime_to_tsobject rather than doing this.

Reply:
We can do this, but it gets appreciably more verbose because we need special handling for datetime vs Timestamp.  The version in the PR ATM is correct for both datetime and Timestmap.